clang/test/CodeGenSYCL/intel-max-global-work-dim.cpp

```diff
@@ -29,17 +26,12 @@ int main() {
 
     Functor<2> f;
     h.single_task<class kernel_name3>(f);
-
```

Comment:
> I can't recall. Have we already removed attribute propagation logic from the compiler i.e. SemaSYCL.cpp. If not, we should.

afaik func() is a device function

So in case std-sycl-2020/default and stack of symbols func()<-kernel_name4 exist and checks NUM2 and kernel_name4 are used the line

./build/bin/llvm-lit -a -v clang/test/CodeGenSYCL/intel-max-global-work-dim.cpp
is producing this

```
build/bin/clang -cc1 -internal-isystem build/lib/clang/19/include -nostdsysteminc -fsycl-is-device -internal-isystem clang/test/CodeGenSYCL/Inputs -triple spir64-unknown-unknown -disable-llvm-passes -emit-llvm -o - clang/test/CodeGenSYCL/intel-max-global-work-dim.cpp | build/bin/FileCheck clang/test/CodeGenSYCL/intel-max-global-work-dim.cpp
+ build/bin/FileCheck clang/test/CodeGenSYCL/intel-max-global-work-dim.cpp
+ build/bin/clang -cc1 -internal-isystem build/lib/clang/19/include -nostdsysteminc -fsycl-is-device -internal-isystem clang/test/CodeGenSYCL/Inputs -triple spir64-unknown-unknown -disable-llvm-passes -emit-llvm -o - clang/test/CodeGenSYCL/intel-max-global-work-dim.cpp
clang/test/CodeGenSYCL/intel-max-global-work-dim.cpp:43:11: error: CHECK: expected string not found in input
```

```
clang/test/CodeGenSYCL/intel-max-global-work-dim.cpp:43:11: error: CHECK: expected string not found in input
// CHECK: define {{.*}}spir_kernel void @{{.*}}kernel_name4() #0 {{.*}} !max_global_work_dim ![[NUM2]]
```


https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#sec:device.attributes

"The C++ attributes that decorate kernels are now better described, and their position has changed so that they are applied directly to the kernel function. (Previously, they were applied to a device function that the kernel calls, and the implementation needed to propagate the information up to the enclosing kernel.) The old C++ attribute form is no longer included in the SYCL specification."

Reply:
I understand. I was saying we have code doing this attribute propagation in SemaSYCL.cpp. That needs to be removed as well as part of this cleanup.

Reply:
> I understand. I was saying we have code doing this attribute propagation in SemaSYCL.cpp. That needs to be removed as well as part of this cleanup.

That will be done in a separate PR. This is a preliminary cleanup to drop 1.2.1 testing and then there will be a follow-up (re-submit of #13411) to drop functionality once we have cleaned up all the tests.

Reply:
I'm ok with removing support in PR #13411 but the current version does not remove full propagation support. So that needs to be modified

Reply:
> but the current version does not remove full propagation support
could you please clarify one more time: version of what?


> So that needs to be modified
could you please clarify one more time: what exactly needs to be modified in this PR?

Reply:
> I'm ok with removing support in PR #13411 but the current version does not remove full propagation support. So that needs to be modified

@elizabethandrews, we intended this PR to be a simple tests update with any compiler code to be removed separately as part of 1.2.1 removal. Do you say that we should start removing the compiler code right away?

Reply:
Sorry for not being clear. I am ok with this PR not removing compiler code. I meant PR #13411 does not remove propagation support from a quick look. So PR #13411 should be modified if that is the PR removing compiler code.

Reply:
My only concern about removing code and tests in different PRs is you will miss compiler code which needs to be removed without failing tests alerting you.

Reply:
I spoke offline with @AndreiZibrov and he mentioned he plans to checkout older versions to run tests to ensure all obsolete code is removed whenever that is done. That is ok with me. So please dismiss my above concern

Reply:
Thats correct, so I hope that the thread is resolved.